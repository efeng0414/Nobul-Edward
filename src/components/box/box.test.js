import React from "react";
import renderer from "react-test-renderer";
import Box from "./index";

describe("Box Component", () => {
  const componentProps = {
    icon: "abc",
    title: "title",
    description: "description",
    onClick: () => {},
    backgroundClass: "",
    className: "component",
    children: "abc",
    intl: global.getIntl()
  };
  const componentJSX = <Box {...componentProps} />;

  describe("Component Snapshot", () => {
    const componentSnapshot = renderer.create(componentJSX).toJSON();
    it("should match existing snapshot", () => {
      expect(componentSnapshot).toMatchSnapshot();
    });
  });
  describe("button click event", () => {
    it("should call onClick function when the button is clicked", () => {
      const componentMount = global.Enzyme.shallow(componentJSX);
      const mockFn = jest.fn();
      componentMount.setProps({ onClick: mockFn });
      componentMount.find("div.box").simulate("click");
      expect(mockFn).toHaveBeenCalled();
    });
  });
});
