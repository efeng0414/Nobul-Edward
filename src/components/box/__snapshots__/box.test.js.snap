// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box Component Component Snapshot should match existing snapshot 1`] = `
<div
  className="box component"
  onClick={[Function]}
>
  <div
    className="box-icon "
  >
    abc
  </div>
  <h2>
    title
  </h2>
  <p>
    description
  </p>
  abc
</div>
`;
